### PR TITLE
Convert OS.path to Pathlib.Path

### PR DESCRIPTION
**Pull request by jhrehanoaa**
augment file has some directory setting do you want me to change that too?

### CODE REVIEW

**Comment by jameshalgren** [comment]
@awlostowski-noaa 
We are making a minor update to the network_augment.py script to use pathlib instead of os.path... what would be the best place to test the output of that script?

**Comment by awlostowski-noaa** [comment]
Y'all, 

First, there is a stale function call at line 90 that needs changed:
```python
network_data = nnu.set_supernetwork_data(
    supernetwork=supernetwork, geo_input_folder=geo_input_folder
)
```
should be...
```python
network_data = nnu.set_supernetwork_parameters(
    supernetwork=supernetwork, geo_input_folder=geo_input_folder
```

Now, you can test the script with the following command line arguments
```
cd src/python_framework_v02/troute
python nhd_network_augment.py --network Pocono_TEST1

cd ../../../test/input/geo/Channels
# Look in here for two new directories: /RouteLink_Pocono_TEST2 and /RouteLink_Pocnon_TEST2_500m_merge
```

If the script works, it should print shapefiles of original and augmented RouteLinks for the Pocono domain.

**Comment by awlostowski-noaa** [comment]
ping @GregCocks-NOAA. See above comment in this thread for how to call the `network_augment` script. 

modifying slightly for Cape Fear FULL RES (which is the domain of the Route Link file I sent you). This will produce two RouteLinks in shapefile format, one original (unmodified), another augmented for a minimum threshold length of 500m. 

```bash
python nhd_network_augment.py --network Florence_FULL_RES -return_original
```

**Comment by GregCocks-NOAA** [comment]
ping @awlostowski-noaa 
Thanks.... will work dircetly with those
This is what I am seeing in the script <> for <>, as you know:

**Comment by awlostowski-noaa** [comment]
@GregCocks-NOAA 

There are some bugs you will discover. For example, if you don't include `--return_original` in the call, the script may break at line 897 because `RouteLink_domain` is undefined. I plan to clear up some of the known bugs in a forthcoming pull request, then we can rebase your branch accordingly.

File: src/python_framework_v02/troute/nhd_network_augment.py [hunk 1]
```diff
@@ -848,25 +848,25 @@ def main():
...
-        os.path.join(root, "test", "input", "geo", "Channels"),
+        pathlib.Path(root, "test", "input", "geo", "Channels").resolve(),
     )
 
-    dir_path = os.path.join(root, "test", "input", "geo", "Channels", dirname)
-    if not os.path.isdir(dir_path):
-        os.mkdir(dir_path)
+    dir_path = pathlib.Path(root, "test", "input", "geo", "Channels", dirname).resolve()
+    # if not os.path.isdir(dir_path):
+    #     os.mkdir(dir_path)
```
Comment (jameshalgren) [inline src/python_framework_v02/troute/nhd_network_augment.py:856]: - [ ] @jhrehanoaa We'll want to find the way to use pathlib to retain this functionality.

File: src/python_framework_v02/troute/nhd_network_augment.py [hunk 2]
```diff
@@ -877,12 +877,12 @@ def main():
...
-            os.path.join(root, "test", "input", "geo", "Channels"),
+            pathlib.Path(root, "test", "input", "geo", "Channels").resolve(),
         )
 
-        dir_path = os.path.join(root, "test", "input", "geo", "Channels", dirname)
-        if not os.path.isdir(dir_path):
-            os.mkdir(dir_path)
+        dir_path = pathlib.Path(root, "test", "input", "geo", "Channels", dirname).resolve()
+        # if not os.path.isdir(dir_path):
+        #     os.mkdir(dir_path)
```
Comment (jameshalgren) [inline src/python_framework_v02/troute/nhd_network_augment.py:885]: - [ ] @jhrehanoaa Ditto above.

File: src/python_framework_v02/troute/nhd_network_utilities_v02.py [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 from functools import partial
 import troute.nhd_io as nhd_io
 import troute.nhd_network as nhd_network
+import pathlib 
 
+ENV_IS_CL = False
+if ENV_IS_CL:
+    root = pathlib.Path("/", "content", "t-route")
+elif not ENV_IS_CL:
+    root = pathlib.Path("../..").resolve()
```
Comment (jameshalgren) [inline src/python_framework_v02/troute/nhd_network_utilities_v02.py:13]: We can probably drop this -- it was used in the Colab notebooks.

**Comment by jameshalgren** [comment]
Error: 
AttributeError: 'PosixPath' object has no attribute 'endswith' 
python_framework_v02/troute/nhd_io.py:16:    if geo_file_path.endswith(".zip"):
python_framework_v02/troute/nhd_io.py:31:    if geo_file_path.endswith(".nc"):
the geo_file_path used to be a string. 
now it's a posix-path object.

Solution: 
we can stringify that line
or there might be a pathlib way to check that the last characters are .nc or .zip.

**Comment by jameshalgren** [comment]
- [ ] Needs to use a translator to get standard column names for the data so that any dataset can be used
- [ ] Fix date error on only merge. 

Places to start...

**Comment by jameshalgren** [comment]
Closing -- subsumed by #243

**Pull request closed without merging**